subsys/mgmt/mcumgr/smp_bt.c

```diff
@@ -156,6 +159,63 @@ static ssize_t smp_bt_chr_write(struct bt_conn *conn,
 				const void *buf, uint16_t len, uint16_t offset,
 				uint8_t flags)
 {
+#ifdef CONFIG_MCUMGR_SMP_REASSEMBLY_BT
+	int ret;
+	bool started;
+
+	started = (zephyr_smp_reassembly_expected(&smp_bt_transport) >= 0);
+
+	LOG_DBG("started = %s, buf len = %d", started ? "true" : "false", len);
+	LOG_HEXDUMP_DBG(buf, len, "buf = ");
+
+	ret = zephyr_smp_reassembly_collect(&smp_bt_transport, buf, len);
+	LOG_DBG("collect = %d", ret);
+
+	/*
+	 * Collection can only fail due to being unable to allocate new buffer or while receiving
+	 * more data than expected in payload.
+	 */
+	if (!started && ret >= 0) {
+		/* On successful collection of first fragment of a new packet, transport
```

Comment:
On successful collection of the first fragment of a new packet the transport information is always attached. The "user data" is always allocated with a buffer, so at this point the ud pointer will not be NULL.

Reply:
Fixed